source/common/protobuf/utility.cc

```diff
@@ -645,12 +645,18 @@ ProtobufWkt::Value ValueUtil::listValue(const std::vector<ProtobufWkt::Value>& v
 
 namespace {
 
-void validateDuration(const ProtobufWkt::Duration& duration, int64_t max_seconds_value) {
-  if (duration.seconds() < 0 || duration.nanos() < 0) {
-    throw DurationUtil::OutOfRangeException(
-        fmt::format("Expected positive duration: {}", duration.DebugString()));
+absl::Status validateDurationNoThrow(const ProtobufWkt::Duration& duration,
+                                     int64_t max_seconds_value) {
+  if (duration.seconds() < 0 || duration.nanos() < 0 || duration.nanos() > 999999999 ||
+      duration.seconds() > max_seconds_value) {
+    ENVOY_LOG_MISC(warn, "Duration out-of-range: {}", duration.DebugString());
+    return absl::OutOfRangeError("Duration out-of-range");
   }
-  if (duration.nanos() > 999999999 || duration.seconds() > max_seconds_value) {
+  return absl::OkStatus();
+}
+
+void validateDuration(const ProtobufWkt::Duration& duration, int64_t max_seconds_value) {
+  if (!validateDurationNoThrow(duration, max_seconds_value).ok()) {
     throw DurationUtil::OutOfRangeException(
         fmt::format("Duration out-of-range: {}", duration.DebugString()));
```

Comment:
This should probably throw the same message received from the NoThrow function.

Reply:
Something like the following:
```
const auto res = validateDurationNoThrow(duration, max_seconds_value)l
if (!result.ok()) {
  throw DurationUtil::OutOfRangeException(res.message());
}
```

Reply:
done!